File: module4/case_study/case_study_springboot/src/main/java/com/example/case_study_springboot/model/entity/employee/Employee.java
package com.example.case_study_springboot.model.entity.employee;

import com.example.case_study_springboot.model.entity.contract.Contract;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@Entity
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(nullable = false)
    private String employeeName;
    @Column(nullable = false)
    private String birthday;
    @Column(nullable = false)
    private String salary;
    @Column(nullable = false)
    private String phone;
    private String email;
    private String address;
    @ManyToOne
    @JoinColumn(referencedColumnName = "id", nullable = false)
    private Division division;
    @ManyToOne
    @JoinColumn(referencedColumnName = "id", nullable = false)
    private EducationDegree educationDegree;
    @ManyToOne
    @JoinColumn(referencedColumnName = "id", nullable = false)
    private Position position;
    @OneToMany(mappedBy = "employee")
    private List<Contract> contractList;
    private Boolean flag;

    public Employee() {
    }

}

File: module5/case-study/src/app/model/employee/employee.ts
import {Division} from './division';
import {EducationDegree} from './education-degree';
import {Position} from './position';

export interface Employee {
  id: number;
  name: string;
  division: Division;
  educationDegree: EducationDegree;
  position: Position;
  birthday: Date;
  idCard: string;
  salary: string;
  phone: string;
  email: string;
  address: string;
}
